options/posix/generic/syslog-stubs.cpp

```diff
@@ -106,8 +106,8 @@ static void _vsyslog(int priority, const char *message, va_list ap) {
 	strftime(timebuf, sizeof timebuf, "%b %e %T", &tm);
 
 	pid = (log_opt & LOG_PID) ? getpid() : 0;
-	l = snprintf(buf, sizeof buf, "<%d>%s %n%s%s%.0d%s: ",
-		priority, timebuf, &hlen, log_ident, "[" + !pid, pid, "]" + !pid);
+	l = snprintf(buf, sizeof buf, "<%d>%s %n%s%c%.0d%c: ",
+		priority, timebuf, &hlen, log_ident, ('[' + !pid), pid, (']' + !pid));
```

Comment:
This is wrong, the previous code skipped the `[` and `]` (by adding 1 to skip the character and get an empty string) from the output if `!pid` was true.